Sources/EditController.swift

```diff
@@ -224,36 +224,52 @@ extension EditController: EditToolbarDelegate {
-                                               textfieldText: media.title,
-                                               confirmActionTitle: NSLocalizedString("BUTTON_RENAME", comment: ""))
-            presentTextFieldAlert(with: alertInfo, completionHandler: {
-                [weak self] text -> Void in
-                guard text != "" else {
-                    VLCAlertViewController.alertViewManager(title: NSLocalizedString("ERROR_RENAME_FAILED", comment: ""),
-                                                            errorMessage: NSLocalizedString("ERROR_EMPTY_NAME", comment: ""),
-                                                            viewController: (UIApplication.shared.keyWindow?.rootViewController)!)
-                    return
-                }
-                media.updateTitle(text)
-                guard let strongself = self else {
-                    return
-                }
-                strongself.delegate?.editController(editController: strongself, cellforItemAt: indexPath)?.isChecked = false
-                strongself.selectedCellIndexPaths.remove(indexPath)
-                //call until all items are renamed
-                if !strongself.selectedCellIndexPaths.isEmpty {
-                    strongself.editToolbarDidRename(editToolbar)
-                }
-            })
+
+        var mlObjectName = ""
+
+        let mlObject = model.anyfiles[indexPath.row]
+
+        if let media = mlObject as? VLCMLMedia {
+            mlObjectName = media.title
+        } else if let playlist = mlObject as? VLCMLPlaylist {
+            mlObjectName = playlist.name
         }
+
+        // Not using VLCAlertViewController to have more customization in text fields
+        let alertInfo = TextFieldAlertInfo(alertTitle: String(format: NSLocalizedString("RENAME_MEDIA_TO", comment: ""), mlObjectName),
+                                           textfieldText: mlObjectName,
+                                           confirmActionTitle: NSLocalizedString("BUTTON_RENAME", comment: ""))
+        presentTextFieldAlert(with: alertInfo, completionHandler: {
+            [weak self] text -> Void in
+            guard text != "" else {
+                VLCAlertViewController.alertViewManager(title: NSLocalizedString("ERROR_RENAME_FAILED", comment: ""),
+                                                        errorMessage: NSLocalizedString("ERROR_EMPTY_NAME", comment: ""),
+                                                        viewController: (UIApplication.shared.keyWindow?.rootViewController)!)
+                return
+            }
+
+            let mlObject = self?.model.anyfiles[indexPath.row]
+
+            if let media = mlObject as? VLCMLMedia {
+                media.updateTitle(text)
+            } else if let playlist = mlObject as? VLCMLPlaylist {
+                playlist.updateName(text)
+            }
```

Comment:
Not a big fan of the duplication here and above for the name/title.

Suggestions welcome! 🎉

Reply:
you could have a wrapper for playlist that is called updateTitle and calls updateName 🤷‍♀️

Reply:
I agree that this isn't great, for now I will add the assertion as you suggested!